catalog/transformer/catalog-transformer-streaming-impl/src/main/java/org/codice/ddf/transformer/xml/streaming/impl/Gml3ToWktImpl.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.transformer.xml.streaming.impl;
+
+import java.io.InputStream;
+import java.util.Collections;
+
+import javax.xml.bind.JAXBElement;
+
+import org.apache.commons.io.IOUtils;
+import org.codice.ddf.parser.Parser;
+import org.codice.ddf.parser.ParserConfigurator;
+import org.codice.ddf.parser.ParserException;
+import org.codice.ddf.transformer.xml.streaming.Gml3ToWkt;
+import org.jvnet.jaxb2_commons.locator.DefaultRootObjectLocator;
+import org.jvnet.ogc.gml.v_3_1_1.jts.ConversionFailedException;
+import org.jvnet.ogc.gml.v_3_1_1.jts.GML311ToJTSGeometryConverter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.vividsolutions.jts.geom.Geometry;
+import com.vividsolutions.jts.io.WKTWriter;
+
+import net.opengis.gml.v_3_1_1.AbstractGeometryType;
+
+public class Gml3ToWktImpl implements Gml3ToWkt {
+    private final Parser parser;
+
+    private final ParserConfigurator configurator;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(Gml3ToWkt.class);
+
+    public Gml3ToWktImpl(Parser parser) {
+        this.parser = parser;
+        this.configurator = parser.configureParser(
+                Collections.singletonList(AbstractGeometryType.class.getPackage().getName()),
```

Comment:
That is an implementation detail of the parser implementation.  Currently we only have one implementation of the `Parser` interface and it is the `XmlParser` you linked to.  It is already doing the tccl pattern since the XML library it is using needs it.  Users of a `Parser` interface should not have to worry about using that pattern.